### PR TITLE
IGNITE-21268 .NET: Expose metric names as public constants

### PR DESCRIPTION
**Pull request by ptupitsyn**
Add public `MetricNames` class with a name for every metric and usage examples in XMLdoc.

Previously, as a user, you had to look at internal class source code to get a list of available metrics; now it is a part of the public API.

### CODE REVIEW

**Review by isapego — APPROVED** [review]

File: modules/platforms/dotnet/Apache.Ignite/MetricNames.cs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+using System.Collections.Generic;
+
+/// <summary>
+/// Ignite.NET client metrics.
+/// <para />
+/// CLI usage example:
+/// <code>
+/// dotnet-counters monitor --counters Apache.Ignite,System.Runtime --process-id PID
+/// </code>
+/// See https://learn.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-counters for more details.
```
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/platforms/dotnet/Apache.Ignite/MetricNames.cs:29]: Fixed.

File: modules/platforms/dotnet/Apache.Ignite/MetricNames.cs [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+///     Console.WriteLine($"{instrument.Name}: {measurement}"));
+///
+/// meterListener.Start();
+/// </code>
+/// See https://learn.microsoft.com/en-us/dotnet/core/diagnostics/metrics for more details.
+/// </summary>
+public static class MetricNames
+{
+    /// <summary>
+    /// Meter name.
```
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/platforms/dotnet/Apache.Ignite/MetricNames.cs:57]: It is a term from the standard library: https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics.meter?view=net-8.0

File: modules/platforms/dotnet/Apache.Ignite/MetricNames.cs [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    /// </summary>
+    public const string HandshakesFailed = "handshakes-failed";
+
+    /// <summary>
+    /// Total number of failed handshakes due to network timeout.
+    /// </summary>
+    public const string HandshakesFailedTimeout = "handshakes-failed-timeout";
+
+    /// <summary>
+    /// Currently active requests (being sent to the socket or waiting for response).
```
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/platforms/dotnet/Apache.Ignite/MetricNames.cs:97]: Fixed.

File: modules/platforms/dotnet/Apache.Ignite/MetricNames.cs [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    /// </summary>
+    public const string RequestsSent = "requests-sent";
+
+    /// <summary>
+    /// Total number of requests completed (response received).
+    /// </summary>
+    public const string RequestsCompleted = "requests-completed";
+
+    /// <summary>
+    /// Total number of request retries.
```
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/platforms/dotnet/Apache.Ignite/MetricNames.cs:112]: It is a number of retry attempts. For example, a given request failed 3 times and then succeeded:
* Total number of retried requests = 1
* Total number of retries = 3

File: modules/platforms/dotnet/Apache.Ignite/MetricNames.cs [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    /// </summary>
+    public const string RequestsRetried = "requests-retried";
+
+    /// <summary>
+    /// Total number of failed requests (failed to send, or completed with error).
```
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/platforms/dotnet/Apache.Ignite/MetricNames.cs:117]: Fixed.

File: modules/platforms/dotnet/Apache.Ignite/MetricNames.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (IgGusev) [reply modules/platforms/dotnet/Apache.Ignite/MetricNames.cs:112]: Lets make it "Total number of retried requests" then. Its a bit ambiguous now.

**Review by IgGusev — APPROVED** [review]

File: modules/platforms/dotnet/Apache.Ignite/MetricNames.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/platforms/dotnet/Apache.Ignite/MetricNames.cs]: Fixed.

**Review by gurustron — APPROVED** [review]
LGTM

File: modules/platforms/dotnet/Apache.Ignite/Internal/Metrics.cs [hunk 6]
```diff
@@ -44,129 +44,129 @@ internal static class Metrics
...
         description: "Total number of lost connections due to network timeout");
 
     /// <summary>
     /// Handshakes failed.
     /// </summary>
     public static readonly Counter<long> HandshakesFailed = Meter.CreateCounter<long>(
-        name: "handshakes-failed",
+        name: MetricNames.HandshakesFailed,
         unit: "handshakes",
         description: "Total number of failed handshakes (due to version mismatch, auth failure, etc)");
```
Comment (gurustron) [inline modules/platforms/dotnet/Apache.Ignite/Internal/Metrics.cs:82]: `etc` -> `etc.`

File: modules/platforms/dotnet/Apache.Ignite/Internal/Metrics.cs [hunk 7]
```diff
@@ -44,129 +44,129 @@ internal static class Metrics
...
 
     /// <summary>
     /// Requests failed.
     /// </summary>
     public static readonly Counter<long> RequestsFailed = Meter.CreateCounter<long>(
-        name: "requests-failed",
+        name: MetricNames.RequestsFailed,
         unit: "requests",
-        description: "Total number of failed requests (failed to send, or completed with error)");
+        description: "Total number of failed requests (due to failure to send, or completed with error)");
```
Comment (gurustron) [inline modules/platforms/dotnet/Apache.Ignite/Internal/Metrics.cs:131]: is comma required here?

**Review by gurustron — APPROVED** [review]

File: modules/platforms/dotnet/Apache.Ignite/Internal/Metrics.cs (hunk 6 above)
Comment (ptupitsyn) [reply modules/platforms/dotnet/Apache.Ignite/Internal/Metrics.cs]: Fixed.

File: modules/platforms/dotnet/Apache.Ignite/Internal/Metrics.cs (hunk 7 above)
Comment (ptupitsyn) [reply modules/platforms/dotnet/Apache.Ignite/Internal/Metrics.cs]: Fixed.

**Pull request merged**